### ISSUE TITLE
Bogus error reported if switch() is not the top-level instruction

### ISSUE THREAD

**Issue by axos88-da** [digital-preservation/csv-validator#299]
For a rule, such as `switch(...) and unique(...)`, 

```
col1:                      switch(($col2/is("VAL1"), starts("1") or starts("2")), ($col2/is("VAL2"), starts("C")), ($col2/is("VAL3"), starts("N"))) and regex("[12CN][a-zA-Z0-9]{9}") and unique($col1, $col3)ƒ
```

the reported error is entirely bogus:
```
(List(s, t, a, r, t, s, (, ", 1, ", ),  , o, r,  , s, t, a, r, t, s, (, ", 2, ", ), s, t, a, r, t, s, (, ", C, ", ), s, t, a, r, t, s, (, ", N, ", ))) and regex("[12CN][a-zA-Z0-9]{9}") and unique( fails for line: 1, column: col2, value: "1bc***"
```

The workaround is to include the other validation in all cases, but that means duplicating code:

```
col1:                      switch(($col2/is("VAL1"), regex("[12][a-zA-Z0-9]{9}") and unique($col1, $col3)), ($col2/is("VAL2"), regex("C[a-zA-Z0-9]{9}") and unique($col1, $col3)), ($col2/is("VAL3"), regex("N[a-zA-Z0-9]{9}") and unique($col1, $col3)))```

**Comment by luketebbs** [comment]
https://github.com/digital-preservation/csv-validator/pull/363
